includes/class-wc-amazon-payments-advanced-alexa-notifications.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Amazon Alexa Notifications class.
+ *
+ * @package WC_Gateway_Amazon_Pay
+ */
+
+/**
+ * Amazon Alexa Notifications class
+ */
+class WC_Amazon_Payments_Advanced_Alexa_Notifications {
+
+	/**
+	 * Constructor, which registers this instance's enable_alexa_notifications_for_carrier
+	 * to the $action provided.
+	 *
+	 * You can integrate Alexa Notifications in your shipping plugin by calling.
+	 * do_action( 'apa_enable_alexa_notifications', $tracking_number, $carrier, $order_id );
+	 *
+	 */
+	public function __construct() {
+		add_action(
+			'apa_enable_alexa_notifications',
+			array( $this, 'enable_alexa_notifications_for_carrier' ),
+			/* Allow third party plugins to alter the priority of this action. */
+			apply_filters( 'apa_enable_alexa_notifications_for_carrier_priority', 10 ),
```

Comment:
This looks like an overkill. What would be a scenario where someone needs this? If they do hook more actions on our action, then they can adjust the priority of those.